### PR TITLE
Lyn 17890 tiaf dev docs

### PR DESCRIPTION
**Pull request by jonawals**
### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Comment by chanmosq** [comment]
`_index.md` files are reserved as landing pages for a directory section, and directories are only needed to define a section that contains a collection of docs. Therefore, you should restructure and rename the files to match the following: 
```
- engine-dev/tools/tiaf/
  - _index.md
  - architectural-overview/
    - _index.md
    - ar-integration.md
    - runtime-frontend/
      - _index.md
      - frontend.md (if only one file, don't need a directory)
      - runtime/
        - _index.md
        - artifact.md
        - build-target.md
        - dependency.md
        - process.md
        - target.md
        - test-engine.md
        - test-runner.md
  - extending-tiaf.md (if only one file, don't need a directory)
  - maintaining.md (if only one file, don't need a directory)
```

**Review by chanmosq — CHANGES_REQUESTED** [review]

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/build-target/_index.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+linkTitle: Build Targets
+title: Build Targets
+description: Information about the TIAF build targets.
+---
+
+The build targets are a higher level abstraction of the [targets]() that treat the individual test and produciton target types as a single, generic build target. Each build target contains a [variant]() that holds either a test target or production target and contains the dependency graph of build targets for that build target. This dependency graph can be walked to discover the build targets that are linked to a given build target.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/build-target/_index.md]: Out of date 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+linkTitle: Runtime & Front End
+title: Runtime & Front End
+description: Information about the TIAF Runtime and Front End.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md:4]: Fixed 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/_index.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+linkTitle: Architectural Overview
+title: Architectural Overview
+description: Architectural overview of the TIAF runtime, front end and Automated Review integration.
+---
+
+The architecture of the TIAF is broadly broken down into two areas: the [Automated Review Integration](./ar-integration), which concerns the integration of the TIAF runtime binaries into O3DE's automated review pipeline, and the [Runtime & Front End](./runtime-frontend), which concerns the architecture of the runtime and front end binaries that run the tests and present the results to the user.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/_index.md:7]: Fixed 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+The runtime and front end work together to present a unified interface to the user for selecting, prioritizing and running tests before presenting a report of the test results to the user. The runtime is responsible for performing the heavy lifting whereas the front end communicates the desired actions to the runtime and presents the results to the user.
+
+Below is a simplified diagram of the ssubsystems that make up the runtime and front end, and how they interact with one another:
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md]: Fixed 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+---
+linkTitle: Runtime & Front End
+title: Runtime & Front End
+description: Information about the TIAF Runtime and Front End.
+---
+
+The runtime and front end work together to present a unified interface to the user for selecting, prioritizing and running tests before presenting a report of the test results to the user. The runtime is responsible for performing the heavy lifting whereas the front end communicates the desired actions to the runtime and presents the results to the user.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md:7]: Fixed 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/dependency/_index.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+---
+linkTitle: Dynamic Dependencies
+title: Dynamic Dependencies
+description: Information about the TIAF dynamic dependencies.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/dependency/_index.md]: Out of date 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/process/_index.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+---
+linkTitle: Process Invocation
+title: Process Invocation
+description: Information about the TIAF process invocation and sheduling.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/process/_index.md]: Out of date 👍

File: content/docs/engine-dev/tools/tiaf/maintaining-tiaf/_index.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+linkTitle: Maintaining
+title: Maintaining the Test Impact Analysis Framework
+description: Information as to how to maintain the TIAF.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/maintaining-tiaf/_index.md:4]: Fixed 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/dependency/_index.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+---
+
+## Dynamic Dependency Map
+
+## Test Selector & Prioritizor
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/dependency/_index.md]: I agree with your suggested naming but it's actually the name of a class (`TestSelectorAndPrioritizor`). I will use that name (with the code markdown) to make it clearer it's a class name 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/target/_index.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+---
+linkTitle: Targets
+title: Targets
+description: Information about the TIAF targets.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/target/_index.md]: Out of date 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/test-engine/_index.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+---
+linkTitle: Test Engine
+title: Test Engine
+description: Information about the TIAF test engine.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/test-engine/_index.md]: Out of date 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/_index.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+---
+linkTitle: Runtime
+title: Runtime
+description: Information about the TIAF Runtime.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/runtime/_index.md]: Out of date 👍

File: content/docs/engine-dev/tools/tiaf/architectural-overview/_index.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+
+## Topics
+
+| Name | Description |
+|-|-|
+| [Automated Review Integration](./ar-integration) | Learn about the architecture of the TIAF integration into O3DE's Automated Review.|
+| [Runtime & Front End](./runtime-frontend) | Learn about the architecture of the TIAF runtime and front end.|
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/tiaf/architectural-overview/_index.md:14]: Fixed 👍

File: content/docs/engine-dev/tools/_index.md [hunk 14]
```diff
@@ -16,3 +16,4 @@ This is a stub page intended to serve as an example of guide structure. Informat
 | Name | Description |
 |-|-|
 | [Project Manager](./project-manager) | Learn about the internals of **Project Manager**, including an overview of the major classes and data structures that work together to create the UI and implement the project management and Gem library features of this tool. |
+| [Test Impact Analysis Framework](./tiaf) | Learn about the internals of the **Test Impact Analysis Framework**, including an overview of the major classes and data structures that work together to create the C++ and Python test support of this tool. |
```
[the remark this answers is not in the document]
Comment (jonawals) [reply content/docs/engine-dev/tools/_index.md:19]: Fixed +1

**Comment by jonawals** [comment]
Structure suggestion from https://github.com/o3de/o3de.org/pull/2338#issuecomment-1495004886 has been fixed 👍

**Pull request closed without merging**